packages/react-devtools-shared/src/devtools/views/Profiler/CommitFlamegraph.js

```diff
@@ -121,10 +128,22 @@ function CommitFlamegraph({chartData, commitTree, height, width}: Props) {
     return null;
   }, [chartData, selectedFiberID, selectedChartNodeIndex]);
 
+  const handleElementMouseEnter = ({id, name}) => {
+    highlightNativeElement(id); // Highlight last hovered element.
+    setHoveredFiberData({id, name}); // Set hovered fiber data for tooltip
+  };
+
+  const handleElementMouseLeave = () => {
+    clearNativeElementHighlight(); // clear highlighting of element on mouse leave
+    setHoveredFiberData(null); // clear hovered fiber data for tooltip
+  };
+
   const itemData = useMemo<ItemData>(
     () => ({
       chartData,
-      hoverFiber,
+      isHovered: hoveredFiberData && hoveredFiberData.id,
+      onElementMouseEnter: handleElementMouseEnter,
+      onElementMouseLeave: handleElementMouseLeave,
```

Comment:
This looks wrong. Neither `handleElementMouseEnter` nor `handleElementMouseLeave` are mentioned in the dependencies array.

I wonder why our lint rule isn't complaining about this.

Reply:
Good catch